src/server/stream_family.cc

```diff
@@ -1444,7 +1476,7 @@ void DestroyGroup(string_view key, string_view gname, ConnectionContext* cntx) {
     return OpDestroyGroup(t->GetOpArgs(shard), key, gname);
   };
 
-  OpStatus result = cntx->transaction->ScheduleSingleHop(std::move(cb));
+  OpStatus result = cntx->transaction->ScheduleSingleHop(cb);
```

Comment:
:100: for making the changes across the file

Reply:
question, what's the reason for removing `move`.

Reply:
I think that the claim is that it doesn't do anything (because all captures are by ref, and so "moving" pointers is the same as copying them).
I would, however, counter-claim that it's a better practice to use `move` even in such cases, because there's no harm, and it's future proof and does not require carefully reviewing the capture list..

Reply:
I have not seen the original request but +1 to what @chakaz  said. If tomorrow `cb` will capture a 1MB string, it will be copied, and performance bugs are hard to catch.

Reply:
ok, I'm good with either way. I will reverse these changes.

Reply:
Ok ok I don't want to drag this, neither do I want @theyueli to reverse changes a third time but  using `move` on a `cb` that does not `move` anything is an anti-pattern.

1. It doesn't help me read the code. Simply put, if there is nothing to be moved, don't move it. Why are you casting something to an R value reference if you don't want that constructor to be used? (It's also a waste, if your class doesn't provide a move cotor/assignment operator)

> I would, however, counter-claim that it's a better practice to use move even in such cases, because there's no harm, and it's future proof and does not require carefully reviewing the capture list..

First, there is a harm, you are stressing the compiler: https://www.foonathan.net/2020/09/move-forward/

I quote

```
Third – and this is the main reason I dislike it – they have compile-time implications. I’m currently working on a library that makes heavy use of meta programming, which already increases compile-times a lot. Still, I can compile the entire test suite in about five seconds (~12K lines). If I were to start using std::move and std::forward, I first need to include <utility> where they’re defined (the majority of the headers don’t need anything besides <type_traits>, <cstddef> etc.). An empty C++ file that just #include <utility> takes 250ms (i.e. 5% of my test suite compilation time) and pulls in about 3K lines of code. Add to that the cost of name lookup, overload resolution and template instantiation every time I want to use them, and compilation times increase by an additional 50ms.
```
(including utility is roughly like including 3k code...)

Also consider, that high caliber libs in C++ use `static_cast` instead of `std::move` :stuck_out_tongue:  Three examples are: ranges-v3, libunifex, and I think folly as well. 

Second, if that's the case why don't we use `std::move` everywhere even on POD types that are trivially copyable? They might change in the future by adding a string member with `1mb` size. I hope especially this points out the fallacy of using std move everywhere.

Sorry for the long rant :stuck_out_tongue:

Reply:
Small rant vol2, it captures everything by reference, even we added a string, we would need to explicitly copy it by value making it nearly impossible for this to happen...

Reply:
Re/ 1: I'd rather have longer compilation times than runtime. When a stressed compiler will be an issue, we can tackle that (and I doubt that this will be the cause of pain for the compiler). I'd even bet that `<utility>` is included in every (or 99%) of TUs we use (either directly or indirectly).
Re/ 2: I'd argue that:
1. It's way more common to capture more local variables (we all add them, all the time) than to add members to structs, and so it makes sense to have a different rule of thumb
2. If you add a 1mb string to your class, it might make sense to also add a move-constructor (if one is not generated for you)
3. It's much less noisier to move callbacks than to move all structs that are ever assigned :)

Reply:
> Small rant vol2, it captures everything by reference, even if we added a string, we would need to explicitly copy it by value making it nearly impossible for this to happen unless you really want to copy this 1mb string...

Simple `grep`ping found 2 cases in which we capture by-ref by-default, but also some vars by value:
https://github.com/romange/helio/blob/937652d5803596893aa774206b36d7b91dccf883/util/fibers/fiberqueue_threadpool.h#L108
https://github.com/romange/helio/blob/937652d5803596893aa774206b36d7b91dccf883/util/fibers/fiberqueue_threadpool.h#L122

Reply:
> If you add a 1mb string to your class, it might make sense to also add a move-constructor (if one is not generated for you) yes but to trigger it you need to call `std::move` everywhere -- ouch! 

> Simple grepping found 2 cases in which we capture by-ref by-default, but also some vars by value:
Both of them capture a `Done` object which internally holds an `intrusive_ptr`. So no deep copying whatsoever...

Reply:
Sorry Shahar, I really don't see any value in using `std::move()` like that. It's noise, even for (1) above, if you add a non trivial type either move it or capture it by reference (the same you would do for any object IMO). Again, since you and Roman agree, I think we got a consensus. I don't want to push staff neither do I think this move is the end of the world or high in importance.